File: Two-Sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map: HashMap<Int, Int> = HashMap()

        for (i in nums.indices) {
            val difference = target - nums[i]
            if (map.containsKey(difference)) {
                return intArrayOf(map[difference]!!, i)
            }
            map[nums[i]] = i
        }
        return intArrayOf()
    }
}

fun main() {
    val nums1 = intArrayOf(2, 7, 11, 15)
    val target1 = 9
    val solution1 = Solution()
    val result1 = solution1.twoSum(nums1, target1)
    println(result1.joinToString()) // 0, 1
}

File: Two-Sum/two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        hashmap = {}
        for i, num in enumerate(nums) :
            difference = target - num
            if difference in hashmap :
                return [hashmap[difference], i]
            hashmap[num] = i
        return []
        


print("True" if Solution().twoSum([2,7,11,15], 13) == [0,2] else "False")
print("True" if Solution().twoSum([3,2,4], 6) == [1,2] else "False")
print("True" if Solution().twoSum([3,3], 6) == [0,1] else "False")
